manager/container.go

```diff
@@ -117,6 +120,34 @@ func (c *containerData) allowErrorLogging() bool {
 	return false
 }
 
+// OnDemandHousekeeping performs housekeeping on the container and blocks until it has completed.
+// It is designed to be used in conjunction with periodic housekeeping, and will cause the timer for
+// periodic housekeeping to reset.  This should be used sparingly, as calling OnDemandHousekeeping frequently
+// can have serious performance costs.
+func (c *containerData) OnDemandHousekeeping(maxAge time.Duration) {
+	if time.Since(c.lastUpdatedTime) > maxAge {
+		housekeepingFinishedChan := make(chan struct{})
+		c.onDemandChan <- housekeepingFinishedChan
+		select {
+		case <-c.stop:
+			close(housekeepingFinishedChan)
+		case <-housekeepingFinishedChan:
+		}
+	}
+}
+
+// notifyOnDemand notifies all calls to OnDemandHousekeeping
```

Comment:
nit: add "that housekeeping is finished"

Reply:
done